dev/tests/integration/testsuite/Magento/Framework/Code/Generator/AutoloaderTest.php

```diff
@@ -0,0 +1,77 @@
+<?php declare(strict_types=1);
+
+namespace Magento\Framework\Code\Test\Unit\Generator;
+
+use Magento\Framework\Code\Generator;
+use Magento\Framework\Code\Generator\Autoloader;
+use Magento\Framework\Logger\Monolog as MagentoMonologLogger;
+use Magento\TestFramework\ObjectManager;
+use PHPUnit\Framework\TestCase;
+use PHPUnit_Framework_MockObject_MockObject as MockObject;
+use Psr\Log\LoggerInterface;
+
+class AutoloaderTest extends TestCase
+{
+    /**
+     * @return ObjectManager|\Magento\Framework\App\ObjectManager
+     */
+    private function getTestFrameworkObjectManager()
+    {
+        return ObjectManager::getInstance();
+    }
+
+    /**
+     * @before
+     */
+    public function setupLoggerTestDouble(): void
+    {
+        $loggerTestDouble = $this->createMock(LoggerInterface::class);
+        $this->getTestFrameworkObjectManager()->addSharedInstance($loggerTestDouble, MagentoMonologLogger::class);
+    }
+
+    /**
+     * @after
+     */
+    public function removeLoggerTestDouble(): void
```

Comment:
I prefer not to override parent class methods. The `@before` and `@after` annotations on public methods with descriptive names are preferable.

Reply:
Such name does not add any value compared to just `setUp`. I have nothing against such annotations in general but then we should make it an official recommendation, otherwise the code become inconsistent.